File: src/de/ax/powermode/power/ElementOfPower.scala
package de.ax.powermode.power

import java.awt.Graphics

import de.ax.powermode.Power

/**
  * Created by nyxos on 10.03.16.
  */
trait ElementOfPower extends Power {

  def update(delta: Float): Boolean

  def render(g: Graphics, dxx: Int, dyy: Int)

  def life: Long

  def initLife: Long

  def lifeFactor: Float = {
    1 - ((life - System.currentTimeMillis()) / initLife.toFloat) toFloat
  }

  def alive: Boolean = {
    life > System
      .currentTimeMillis() && powerMode.isEnabled //&& width >= 0 && width <= _width
  }
}

File: src/com/nmeylan/powermode/power/ElementOfPower.java
package com.nmeylan.powermode.power;


import com.nmeylan.powermode.Power;

import java.awt.Graphics;

public interface ElementOfPower extends Power {

  boolean update(double delta);

  void render(Graphics g, int dxx, int dyy);

  long life();

  long initLife();

  default float lifeFactor() {
    return 1 - ((life() - System.currentTimeMillis()) / (float)initLife());
  }
  default boolean alive() {
    return life() > System.currentTimeMillis() && powerMode().isEnabled();
  }
}
